### ISSUE TITLE
Shebang detection fail

### ISSUE THREAD

**Issue by th4tkh13m** [nathom/filetype.nvim#59]
When I open a file with shebang, I get the following error:
```
Error detected while processing BufReadPost Autocommands for "*":
E5108: Error executing lua ...vim/site/pack/packer/opt/packer.nvim/lua/packer/load.lua:165: Vim(echomsg):E114: Missing quote: "Error in packer_compiled: ...vim/site/pack/packer/opt/packer.nvim/lua/packer/load.lua:155: Vim(lua):E5108: Error executing lua ...te/pack/packer/start/filetype.nvim/lua/filetype/init.lua:183:
 attempt to index upvalue 'custom_map' (a nil value)
stack traceback:
        [C]: in function 'cmd'
        ...vim/site/pack/packer/opt/packer.nvim/lua/packer/load.lua:165: in function <...vim/site/pack/packer/opt/packer.nvim/lua/packer/load.lua:161>
        [string ":lua"]:1: in main chunk
E5108: Error executing lua ...te/pack/packer/start/filetype.nvim/lua/filetype/init.lua:183: attempt to index upvalue 'custom_map' (a nil value)
stack traceback:
        ...te/pack/packer/start/filetype.nvim/lua/filetype/init.lua:183: in function 'resolve'
        [string ":lua"]:1: in main chunk
```
I'm using NeoVim 0.6 but this issue occurs in 0.5 too.

Note that this only occurs with some files only, in particular:
- A new file with shebang only (I created a file named `test`, with the content `#!\bin\sh`) 
- Some system file with shebang (`finish` and `run` file for services in Void Linux)
For example: `finish` file:
```
#!/bin/sh
set -e
exec alsactl store
```

**Comment by clason** [comment]
See https://github.com/nathom/filetype.nvim/pull/56

**Comment by mohamed-abdelnour** [comment]
Hello, everyone. Even after #56 this wouldn't work properly on the `main` branch.

https://github.com/nathom/filetype.nvim/blob/0426dc8fee6d9b49b5895206c3c02f65135d25bc/lua/filetype/init.lua#L184

On the `main` branch, shebangs are only applied from `custom_map`. This means that unless you configure a custom override for `sh` shebangs, the file type wouldn't get set to `sh`.

This has been fixed on the `dev` branch in #53. You can use the switch to using `dev` in the meantime until it gets merged into `main`.
